### PR TITLE
clear error for unauthenticated request

### LINKED ISSUE #101: Show more clear error when token cannot query some API
If we get a 401 due to using a token that doesn't have permissions for that API, we should show a better error. Here's what we currently show:

```bash
$ ipinfo AS63989
err: GET https://ipinfo.io/AS63989: 401 { }
```

We should show an error similar to the one shown when you have no token at all:

```bash
$ ipinfo logout
logged out

$ ipinfo AS63989
err: ASN lookups require a token; login via `ipinfo login`.
```

### PR DESCRIPTION
**Pull request by rm-Umar**
fixed #101

### CODE REVIEW

File: ipinfo/cmd_asn.go [hunk 1]
```diff
@@ -67,6 +68,9 @@ func cmdASN(asn string) error {
 
 	data, err := ii.GetASNDetails(asn)
 	if err != nil {
+		if strings.Contains(err.Error(), "401 { }") {
```
Comment (UmanShahzad) [inline ipinfo/cmd_asn.go:71]: It'd be nice if we could get at the underlying error type and refer to it directly. Parsing the error string like this definitely feels like a hack. Unless the underlying error type is storing any important info as a string only and no other form.

Comment (rm-Umar) [reply ipinfo/cmd_asn.go:71]: The underlying error is just a string without any additional information that we can use. that's why this was the only possible solution I could figure out. 
```
string
err: GET https://ipinfo.io/AS63987: 401 { }
```

Comment (UmanShahzad) [reply ipinfo/cmd_asn.go:71]: See the IPinfo Go SDK. Follow the logic carefully from the `GetASNDetails` function call and you will find the actual error type. It is a real error type and contains even the full Response object in it.

**Review by UmanShahzad — APPROVED** [review]

**Review by appleboy159 — COMMENTED** [review]
fix

**Pull request merged**